backend-lib/src/main/rust/lib.rs

```diff
@@ -101,9 +101,9 @@ fn account_id_from_jint(account: jint) -> Result<AccountId, failure::Error> {
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_initOnLoad(
-    _env: JNIEnv<'_>,
-    _: JClass<'_>,
+pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_initOnLoad<'local>(
```

Comment:
The `jni` crate docs use `extern "system"`, which is correct.

On Unix it will work with `extern "C"`, but that's wrong in general. Specifically, on Win32 `JNICALL` declares the `stdcall` convention, not the `C` convention. `extern "system"` works on all platforms.

I know Windows is currently a Tier 3 platform. Nevertheless, please replace all `extern "C"`s for JNI native method definitions with `extern "system"`. This was already incorrect for `jni 0.20` so it could be done in a separate PR if that's easier.

Reply:
There are now Android Windows targets. The host is irrelevant here; what matters is the target being compiled for, which per the available Android targets and the definition of `system` means that it will always be equivalent to `C`.

Reply:
Opened #1358.